code/game/objects/items/devices/uplinks.dm

```diff
@@ -14,6 +14,7 @@ A list of items and costs is stored under the datum of every game mode, alongsid
 	var/show_description = null
 	var/active = 0
 	var/job = null
+	var/datum/role/owner //The original owner of this uplink
```

Comment:
Not sure how critical it is (as roles don't get qdel'd often), but you could make it so this reference is properly cleared a role is deleted

Reply:
This was used in an outdated version before I reconsidered and went with GetRole, I no longer use this owner ref

I'll delete it